tests/test_html_storage_middleware.py

```diff
@@ -19,22 +19,43 @@ def make_request_mock(save_html=False, query='', results_page=None):
 
     return request_mock
 
+def make_response_mock(response_status):
+    """ Constructs HTTP Response mock object.
+    """
+    response_mock = MagicMock()
+    response_mock.status = response_status
+
+    return response_mock
 
-def test_should_save_html_returns_true_when_request_metainformation_has_special_key_set():
+
+@pytest.mark.parametrize('response_status', [200, 404])
+def test_should_save_html_returns_true_when_request_metainformation_has_special_key_set_and_appropriate_response_status(response_status):
     request_mock = make_request_mock(save_html=True)
+    response_mock = make_response_mock(response_status=response_status)
 
-    save = should_save_html(request_mock)
+    save = should_save_html(request_mock, response_mock)
 
     assert_that(save, is_(True))
 
 
+@pytest.mark.parametrize('response_status', [999, 503])
```

Comment:
These are new tests.
So they are not supposed to be commited with "Adjust tests" changes.
These tests should be commited together with https://github.com/povilasb/scrapy-html-storage/pull/2/commits/0dd89ff19e00c56487401757557483e6b90446d6